### PR TITLE
hoon: implement signed int parsing

### PR DESCRIPTION
**Pull request by pkova**
Needed this for advent of code day 5, seems useful enough.

### CODE REVIEW

File: pkg/arvo/sys/hoon.hoon [hunk 1]
```diff
@@ -5605,6 +5605,15 @@
 ++  but  (cook |=(a/@ (sub a '0')) (shim '0' '1'))      ::  binary digit
 ++  cit  (cook |=(a/@ (sub a '0')) (shim '0' '7'))      ::  octal digit
 ++  dem  (bass 10 (most gon dit))                       ::  decimal to atom
+++  dsm                                                 ::  decimal to signed int
+  %+  cook
+    |=  [s=? n=@]
+    ^-  @s
+    ?:  s
+      (mul n 2)
+    (dec (mul n 2))
```
Comment (Fang-) [inline pkg/arvo/sys/hoon.hoon]: You can replace this with just `new:si`, which is the existing stdlib function for "unsigned to signed", and matches the function signature used here.

File: pkg/arvo/sys/hoon.hoon [hunk 2]
```diff
@@ -5605,6 +5605,15 @@
 ++  but  (cook |=(a/@ (sub a '0')) (shim '0' '1'))      ::  binary digit
 ++  cit  (cook |=(a/@ (sub a '0')) (shim '0' '7'))      ::  octal digit
 ++  dem  (bass 10 (most gon dit))                       ::  decimal to atom
+++  dsm                                                 ::  decimal to signed int
```
Comment (Fang-) [inline pkg/arvo/sys/hoon.hoon]: Might be nice if we can maintain the `cvc` style naming here. Maybe `sem`?

**Comment by ohAitch** [comment]
Yeah this definitely goes into a library not hoon.hoon: 4i is for lexing
/the hoon grammar/ specifically

**Comment by pkova** [comment]
Thanks guys, I'll be back when I figure out where to put this.

`@s` parsing seems to be done in `tash:so`.

**Pull request closed without merging**